### PR TITLE
Create 23.10.1.md Release Notes

### PR DESCRIPTION
**Pull request by Ulrick28**
Release notes and features for 23.10.1 release.

### CODE REVIEW

File: releases/23.10.1.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+The **Open 3D Engine (O3DE)** 23.10.1 fully implements Prefab Overrides and the Document Property Editor (DPE), enabling these by default. The release also simplifies steps for exporting packages, fixes crashes in the Material Canvas, implements Asset Browser optimizations to improve editor startup times in projects with large numbers of assets (eg several thousand assets), and ROS2 gem and Robotics Templates bug fixes and improvements.
+
+## Highlights
+
+Here are some highlights of O3DE 23.10, followed by a detailed list of features broken down by Special Interest Group (SIG).
```
Comment (AMZN-daimini) [inline releases/23.10.1.md:5]: Should be 23.10.1?

File: releases/23.10.1.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+The **Open 3D Engine (O3DE)** 23.10.1 fully implements Prefab Overrides and the Document Property Editor (DPE), enabling these by default. The release also simplifies steps for exporting packages, fixes crashes in the Material Canvas, implements Asset Browser optimizations to improve editor startup times in projects with large numbers of assets (eg several thousand assets), and ROS2 gem and Robotics Templates bug fixes and improvements.
```
Comment (AMZN-daimini) [inline releases/23.10.1.md:1]: nit: eg -> e.g.

File: releases/23.10.1.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+- Material canvas base and standard PBR nodes now use the material pipeline system to share shader code, generate fewer files, and look closer to the original material types. Material graph files will not need to be updated but users might need to delete previously generated material and shader files and let material canvas create new ones. It might take longer to process assets and update previews depending on how many material pipelines, platforms, and RHI are enabled. These can be configured in project and asset processor registry settings.
+- Fixed material and shader builder warnings, errors, dependency tracking, and problems with consistent reloading that affected iterating on shaders and material types in and out of material canvas.
+- Material component generate source materials feature now tracks and displays progress for generated materials to ensure that assets are ready before assigning them. This would previously print several warning messages when assigning unprocessed materials.
+- Material component, instance editor, and generate source materials have been fixed to reference original source material types instead generated files in the intermediate asset folder.
```
Comment (AMZN-daimini) [inline releases/23.10.1.md:14]: instead -> instead of

File: releases/23.10.1.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+- The atom instance database and instance id classes were updated with better support for tracking specific versions of assets so that new instances can be easily created when assets reload.
```
Comment (AMZN-daimini) [inline releases/23.10.1.md:15]: Should Atom be capitalized here?

**Review by AMZN-daimini — APPROVED** [review]
Very minor nitpicks, but looks good overall!

**Review by spham-amzn — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by alexmontAmazon — APPROVED** [review]
approved, pending @AMZN-daimini 's nits

**Comment by Ulrick28** [comment]
Addressed all feedback

**Pull request merged**